tests/test_sources.py

```diff
@@ -92,3 +95,56 @@ def test_provides_correct_contents(self, fancy_wheel):
 
         assert sorted(got_records) == sorted(expected_records)
         assert got_files == files
+
+    def modify_wheel_record(self, fancy_wheel, manipulation_func):
+        """Helper function for modifying RECORD in the wheel file.
+
+        Exists because ZipFile doesn't support remove.
+        """
+        files = {}
+        # Read everything except RECORD and add it back immediately
+        with zipfile.ZipFile(fancy_wheel) as archive:
+            for file in archive.namelist():
+                # Call manipulation function so that we can add RECORD back.
+                if file.endswith("RECORD"):
+                    manipulation_func(files, file, archive.read(file))
+                    continue
+
+                files[file] = archive.read(file)
+        # Replace original archive
+        with zipfile.ZipFile(fancy_wheel, mode="w") as archive:
+            for name, content in files.items():
+                archive.writestr(name, content)
+
+    def test_validation_error_no_record(self, fancy_wheel):
```

Comment:
nit: I'd like to follow the current naming style:

```suggestion
    def test_rejects_no_record_on_validate(self, fancy_wheel):
```

```py
    def test_rejects_missing_hash_on_validate(self, fancy_wheel):
```